test/support/postgresql/import_examples.rb

```diff
@@ -127,3 +127,174 @@ def should_support_postgresql_import_functionality
+          should_update_fields_mentioned
+        end
+
+        context "using string hash map" do
+          let(:update_columns){ { "title" => "title", "author_email_address" => "author_email_address", "parent_id" => "parent_id" } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned
+        end
+
+        context "using string hash map, but specifying column mismatches" do
+          let(:update_columns){ { "title" => "author_email_address", "author_email_address" => "title", "parent_id" => "parent_id" } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned_with_hash_mappings
+        end
+
+        context "using symbol hash map" do
+          let(:update_columns){ { :title => :title, :author_email_address => :author_email_address, :parent_id => :parent_id } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned
+        end
+
+        context "using symbol hash map, but specifying column mismatches" do
+          let(:update_columns){ { :title => :author_email_address, :author_email_address => :title, :parent_id => :parent_id } }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned_with_hash_mappings
+        end
+      end
+
+      context "without specifying a unique constraint" do
+        let(:columns){  %w( id title author_name author_email_address parent_id ) }
+        let(:values){ [ [ 100, "Book", "John Doe", "john@doe.com", 17 ] ] }
+        let(:updated_values){ [ [ 100, "Book - 2nd Edition", "This should raise an exception", "john@nogo.com", 57 ] ] }
+
+        macro(:perform_import) do |*opts|
+          Topic.import columns, updated_values, opts.extract_options!.merge(:on_duplicate_key_update => { columns: [:title] }, :validate => false)
+        end
+
+        setup do
+          Topic.import columns, values, :validate => false
+          @topic = Topic.find 100
+        end
+
+        it "should raise an ArgumentError" do
+          assert_raise ArgumentError do
+            perform_import
+          end
+        end
+      end
+
+      context "without specifying update columns" do
+        let(:columns){  %w( id title author_name author_email_address parent_id ) }
+        let(:values){ [ [ 100, "Book", "John Doe", "john@doe.com", 17 ] ] }
+        let(:updated_values){ [ [ 100, "Book - 2nd Edition", "This should raise an exception", "john@nogo.com", 57 ] ] }
+
+        macro(:perform_import) do |*opts|
+          Topic.import columns, updated_values, opts.extract_options!.merge(:on_duplicate_key_update => { key: :id }, :validate => false)
+        end
+
+        setup do
+          Topic.import columns, values, :validate => false
+          @topic = Topic.find 100
+        end
+
+        it "should raise an ArgumentError" do
+          assert_raise ArgumentError do
+            perform_import
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
@zdennis using your suggestion to default `:conflict_target` to the primary key, it is now possible to provide an array to :on_duplicate_key_update for Postgres.

`:on_duplicate_key_update => [:column1, :column2]`

I've made the array option a baseline example and pulled that out into a separate shared examples file. The hash option is included in both of the adapter specific examples.